File: C#/0441. Arranging Coins.cs
public class Solution {
    public int ArrangeCoins(int n) {
        if(n==0||n==1){
            return n;
        }
        long left=0;
        long right=n;
        while (left<=right){
            long mid=left+(right-left)/2;
            long tmp=(1+mid)*mid/2;
            if (tmp==n){
                return (int)mid;
            }
            else if(tmp>n){
                right=mid-1;
            }
            else{
                left=mid+1;
            }
        }
        return (int)right;
        
    }
}

File: 20190519python/0441. Arranging Coins.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        #(1+i)*i//2
        lo=0
        hi=n
        while lo<=hi:
            mid=lo+(hi-lo)//2
            target=(1+mid)*mid//2
            if target==n:
                return mid
            elif target>n:
                hi=mid-1
            else:
                lo=mid+1
        return lo-1
